scripts/csv_bench.sh

```diff
@@ -0,0 +1,115 @@
+usage() {
+    cat <<_end_
+Usage: csv-bench.sh --path=PATH [--coreneuron]
+
+Generate CSV file summarising a set of benchmark runs.
+
+Options:
+    --path=PATH      Path containing the output
+    --coreneuron     If the path contains CoreNeuron output.
+_end_
+    exit 1
+}
+
+parse_coreneuron=false
+path=
+
+while [ "$1" != "" ]
+do
+    case $1 in
+        --coreneuron )
+            parse_coreneuron=true
+            ;;
+        --path=* )
+            path="${1#--path=}"
+            ;;
+        --path )
+            shift
+            path=$1
+            ;;
+        * )
+            echo "unknown option '$1'"
+            usage
+    esac
+    shift
+done
+
+if [ "$path" == "" ]
+then
+    usage
+    exit 1
+fi
+
+if [ ! -d "$path" ]
+then
+    echo "error: path \"$path\" does not exist"
+    exit 1
+fi
+
+table_line() {
+    fid="$1"
+    line=
+
+    tts=$(awk '/^model-run/ {print $2}' "$fid")
+    ncell=$(awk '/^cell stats/ {print $3}' "$fid")
+
+    line=$(printf "%9d,%12.3f," $ncell $tts)
+    nranks=$(awk '/^ranks:/ {print $2}' "$fid")
+
+    mempos=$(awk '/^meter / {j=-1; for(i=1; i<=NF; ++i) if($i =="memory(MB)") j=i; print j}' "$fid")
+    if [ "$mempos" != "-1" ]
+    then
+        rankmem=$(awk "/^meter-total/ {print \$$mempos}" "$fid")
+        totalmem=$(echo $rankmem*$nranks | bc -l)
+        line="$line$(printf "%12.3f," $totalmem)"
```

Comment:
Again we have here the issue that it looks like the quoting is not intentional; `$line$(printf ` is quoted, and ` $totalmen)` is quoted, but …